adapters/rubicon/rubicon.go

```diff
@@ -941,6 +909,144 @@ func resolveBidFloor(bidFloor float64, bidFloorCur string, reqInfo *adapters.Ext
 	return bidFloor, nil
 }
 
+func updateImpRpTargetWithFpdAttributes(extImp openrtb_ext.ExtImpRubicon, imp openrtb2.Imp,
+	site *openrtb2.Site, app *openrtb2.App) (json.RawMessage, error) {
+	existingTarget, _, _, err := jsonparser.Get(imp.Ext, "rp", "target")
+	if err != nil {
+		logParsingError(err)
+	}
+
+	target := populateFirstPartyDataAttributes(extImp.Inventory, existingTarget)
+
+	if site != nil {
+		siteExtData, _, _, err := jsonparser.Get(site.Ext, "data")
+		if err != nil {
+			logParsingError(err)
+		}
+		target = populateFirstPartyDataAttributes(siteExtData, target)
+		if len(site.SectionCat) > 0 {
+			target = addStringArrayAttribute(site.SectionCat, target, "sectioncat")
+		}
+		if len(site.PageCat) > 0 {
+			target = addStringArrayAttribute(site.PageCat, target, "pagecat")
+		}
+		if site.Page != "" {
+			target = addStringAttribute(site.Page, target, "page")
+		}
+		if site.Ref != "" {
+			target = addStringAttribute(site.Ref, target, "ref")
+		}
+		if site.Search != "" {
+			target = addStringAttribute(site.Search, target, "search")
+		}
+	} else {
+		appExtData, _, _, err := jsonparser.Get(app.Ext, "data")
+		if err != nil {
+			logParsingError(err)
+		}
+		target = populateFirstPartyDataAttributes(appExtData, target)
+		if len(app.SectionCat) > 0 {
+			target = addStringArrayAttribute(app.SectionCat, target, "sectioncat")
+		}
+		if len(app.PageCat) > 0 {
+			target = addStringArrayAttribute(app.PageCat, target, "pagecat")
+		}
+	}
+
+	impExtContextAttributes, _, _, _ := jsonparser.Get(imp.Ext, "context", "data")
+	if err != nil {
+		logParsingError(err)
+	}
+	impExtDataAttributes, _, _, _ := jsonparser.Get(imp.Ext, "data")
+	if err != nil {
+		logParsingError(err)
+	}
+
+	if impExtContextAttributes != nil && len(impExtContextAttributes) > 0 {
+		target = populateFirstPartyDataAttributes(impExtContextAttributes, target)
+	} else if impExtDataAttributes != nil && len(impExtDataAttributes) > 0 {
+		target = populateFirstPartyDataAttributes(impExtDataAttributes, target)
+	}
+
+	if len(extImp.Keywords) > 0 {
+		target = addStringArrayAttribute(extImp.Keywords, target, "keywords")
+	}
+
+	return target, nil
+}
+
+func logParsingError(err error) {
+	glog.Infof(fmt.Sprintf("Error while parsing. Error: %s", err.Error()))
```

Comment:
As you wrote `We require that these errors aren't ignored because otherwise it becomes really difficult to debug`. For me this logs is not needed, and errors like `unexpected end of JSON input` should be ignored. What type of errors can be produced here, that should be returned to user? Maybe, I haven't understand you correctly from the beginning, but will appreciate your answers)
(Updated PR, with new logic for some errors)

Reply:
Well, I'd say errors like `unexpected end of JSON input` shouldn't be ignored. They should rather be communicated back to the publisher so that the publisher knows that there's possibly a syntax error in their raw JSON string within the request and they can fix it accordingly.  If you just ignore the errors, publishers won't know about them and these errors might cause lower/no bids from your adapter, for example. And if a publisher then wants to debug why they are getting lower/no bids from Rubicon, it will be very difficult because these errors are just silently being ignored. I. hope that makes sense